src/stream/src/common/table/state_table.rs

```diff
@@ -691,7 +720,27 @@ impl<S: StateStore> StateTable<S> {
                 }
             }
         }
+        if let Some(range_end_suffix) = range_end_suffix {
+            let range_begin_suffix = if let Some(ref last_watermark) = self.last_watermark {
+                serialize_pk(
+                    &Row::new(vec![Some(last_watermark.clone())]),
+                    prefix_serializer.as_ref().unwrap(),
+                )
+            } else {
+                vec![]
+            };
+            for vnode in self.vnodes.ones() {
```

Comment:
Do we need to delete data for every epoch?

Reply:
> Do we need to delete data for every epoch?

fixed